api/examples/kafka-ephemeral-ingress.yaml

```diff
@@ -19,22 +19,241 @@ spec:
           brokers:
           - broker: 0
             host: broker-0.${cluster_name}.${kube_ip}.nip.io
-#          - broker: 1
-#            host: broker-1.${cluster_name}.${kube_ip}.nip.io
+          - broker: 1
+            host: broker-1.${cluster_name}.${kube_ip}.nip.io
+          - broker: 2
+            host: broker-2.${cluster_name}.${kube_ip}.nip.io
+      - name: oauth
+        port: 9093
+        type: ingress
+        tls: true
+        configuration:
+          bootstrap:
+            host: bootstrap-oauth.${cluster_name}.${kube_ip}.nip.io
+            annotations:
+              "eyefloaters.github.com/console-listener": "true"
+          brokers:
+          - broker: 0
+            host: broker-0-oauth.${cluster_name}.${kube_ip}.nip.io
+          - broker: 1
+            host: broker-1-oauth.${cluster_name}.${kube_ip}.nip.io
+          - broker: 2
+            host: broker-2-oauth.${cluster_name}.${kube_ip}.nip.io
+        authentication:
+          jwksEndpointUri: "${keycloak_realm_url}/protocol/openid-connect/certs"
+          type: oauth
+          userNameClaim: preferred_username
+          validIssuerUri: "${keycloak_realm_url}"
     config:
-      offsets.topic.replication.factor: 1
-      transaction.state.log.replication.factor: 1
-      transaction.state.log.min.isr: 1
-      default.replication.factor: 1
-      min.insync.replicas: 1
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 3
+      min.insync.replicas: 2
       inter.broker.protocol.version: "3.5"
       allow.everyone.if.no.acl.found: "true"
     storage:
       type: ephemeral
+    metricsConfig:
+      type: jmxPrometheusExporter
+      valueFrom:
+        configMapKeyRef:
+          name: ${cluster_name}-kafka-metrics
+          key: kafka-metrics-config.yml
   zookeeper:
-    replicas: 1
+    replicas: 3
     storage:
       type: ephemeral
+    metricsConfig:
+      type: jmxPrometheusExporter
+      valueFrom:
+        configMapKeyRef:
+          name: ${cluster_name}-kafka-metrics
+          key: zookeeper-metrics-config.yml
   entityOperator:
     topicOperator: {}
     userOperator: {}
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: ${cluster_name}-kafka-metrics
+  labels:
+    app: strimzi
+data:
+  kafka-metrics-config.yml: |
```

Comment:
If the backend has requirements for how the metrics are named let's add a comment here explaining that that is the case.

Reply:
These metrics configurations aren't actually used by the API (I had them around from some experimentation with metrics I was attempting) so there's nothing significant from a naming perspective here. I figure it's harmless enough to have in this PR and we can document metrics configuration/requirements properly once the Prometheus integration is finally included.